tests/python/test_clang.py

```diff
@@ -917,6 +917,18 @@ def test_probe_read_kprobe_ctx(self):
     sk = (struct sock *)ctx->di;
     return sk->sk_dport;
 }
+"""
+        b = BPF(text=text)
+        fn = b.load_func("test", BPF.KPROBE)
+
+    def test_probe_read_ctx_array(self):
+        text = """
+#include <linux/sched.h>
+#include <net/inet_sock.h>
+int test(struct pt_regs *ctx) {
+    struct sock *newsk = (struct sock *)PT_REGS_RC(ctx);
```

Comment:
The test will fail on top of bcc trunk. The following change can fix the issue
```
diff --git a/src/cc/frontends/clang/b_frontend_action.cc b/src/cc/frontends/clang/b_frontend_action.cc
index 3865343..7691191 100644
--- a/src/cc/frontends/clang/b_frontend_action.cc
+++ b/src/cc/frontends/clang/b_frontend_action.cc
@@ -420,7 +420,7 @@ bool ProbeVisitor::VisitMemberExpr(MemberExpr *E) {
   }
 
   if (!rewriter_.isRewritable(E->getLocStart()))
-    return false;
+    return true;
 
   /* If the base of the dereference is a call to another function, we need to
    * visit that function first to know if a rewrite is necessary (i.e., if the
```
Basically, we do want to terminate the AST traversal. We just want to
skip this one and proceed to the next.
Could you incorporate this change in your next revision so tests can pass.

Reply:
Done.